datafusion/core/src/datasource/statistics.rs

```diff
@@ -165,11 +225,75 @@ pub(crate) fn get_col_stats(
-                min_value,
-                distinct_count: None,
+                null_count: null_counts[i].clone(),
+                max_value: max_value.map(Precision::Exact).unwrap_or(Precision::Absent),
+                min_value: min_value.map(Precision::Exact).unwrap_or(Precision::Absent),
+                distinct_count: Precision::Absent,
             }
         })
         .collect()
 }
+
+/// If the given value is numerically greater than the original maximum value,
+/// set the new maximum value with appropriate exactness information.
+fn set_max_if_greater(
+    max_values: &mut [Precision<ScalarValue>],
+    max_nominee: Precision<ScalarValue>,
+    index: usize,
+) {
+    match (&max_values[index], &max_nominee) {
+        (Precision::Exact(val1), Precision::Exact(val2)) => {
+            if val1 < val2 {
+                max_values[index] = max_nominee;
+            }
+        }
+        (Precision::Exact(val1), Precision::Inexact(val2))
+        | (Precision::Inexact(val1), Precision::Inexact(val2))
+        | (Precision::Inexact(val1), Precision::Exact(val2)) => {
+            if val1 < val2 {
+                max_values[index] = max_nominee.to_inexact()
+            }
+        }
+        (Precision::Inexact(_), Precision::Absent)
+        | (Precision::Exact(_), Precision::Absent) => {
+            max_values[index] = max_values[index].clone().to_inexact()
+        }
+        (Precision::Absent, Precision::Exact(_))
+        | (Precision::Absent, Precision::Inexact(_)) => {
+            max_values[index] = max_nominee.to_inexact()
+        }
+        (Precision::Absent, Precision::Absent) => max_values[index] = Precision::Absent,
+    }
+}
+
+/// If the given value is numerically lesser than the original minimum value,
+/// set the new minimum value with appropriate exactness information.
+fn set_min_if_lesser(
+    min_values: &mut [Precision<ScalarValue>],
+    min_nominee: Precision<ScalarValue>,
+    index: usize,
+) {
+    match (&min_values[index], &min_nominee) {
+        (Precision::Exact(val1), Precision::Exact(val2)) => {
+            if val1 > val2 {
+                min_values[index] = min_nominee;
+            }
+        }
+        (Precision::Exact(val1), Precision::Inexact(val2))
+        | (Precision::Inexact(val1), Precision::Inexact(val2))
+        | (Precision::Inexact(val1), Precision::Exact(val2)) => {
+            if val1 > val2 {
+                min_values[index] = min_nominee.to_inexact()
+            }
+        }
+        (Precision::Inexact(_), Precision::Absent)
+        | (Precision::Exact(_), Precision::Absent) => {
+            min_values[index] = min_values[index].clone().to_inexact()
+        }
+        (Precision::Absent, Precision::Exact(_))
+        | (Precision::Absent, Precision::Inexact(_)) => {
+            min_values[index] = min_nominee.to_inexact()
+        }
+        (Precision::Absent, Precision::Absent) => min_values[index] = Precision::Absent,
+    }
+}
```

Comment:
It is somewhat worrying that this file has no unit tests 🤔  (not that your PR made this any better / worse)